distribution/docker/build.gradle

```diff
@@ -137,6 +137,7 @@ processTestResources {
 }
 
 task integTest(type: Test) {
+  outputs.doNotCacheIf('Build cache is disabled for Docker tests') { true }
```

Comment:
This isn't actually a packaging test, it's running rest tests with docker. I don't think we disable caching for the rest of rest or qa tests do we?

Reply:
True it's not a packaging test, but the same applies, we aren't considering the Docker image as an input here.

We don't use the Docker image in any other tests, with the exception of those being added in #51658, which prompted me to look into this. For other rest and qa tests we use test clusters, which handles tracking the node distribution as a proper task input.